object.cpp

```diff
@@ -753,16 +753,22 @@ std::string object_description(int obj)
         }
         temp += " to study above level 2.";
     }
-    int buildable = 1;
+    bool buildable;
     auto pS = FindSkill(o->skill);
-    if (o->item == -1 || o->skill == nullptr || !pS || pS->get().flags & SkillType::DISABLED) buildable = 0;
-    if (o->item != I_WOOD_OR_STONE && (ItemDefs[o->item].flags & ItemType::DISABLED)) buildable = 0;
-    if (
-        o->item == I_WOOD_OR_STONE && (ItemDefs[I_WOOD].flags & ItemType::DISABLED) &&
-        (ItemDefs[I_STONE].flags & ItemType::DISABLED)
-    ) {
-        buildable = 0;
+    if (o->item == -1 || o->skill == nullptr || !pS || pS->get().flags & SkillType::DISABLED)
+    {
+        buildable = false;
     }
+    else if (o->item >= 0 && (ItemDefs[o->item].flags & ItemType::DISABLED))
+    {
+        buildable = false;
+    }
+    else {
+        buildable = (o->item == I_WOOD_OR_STONE
+            && (ItemDefs[I_WOOD].flags & ItemType::DISABLED)
+            && (ItemDefs[I_STONE].flags & ItemType::DISABLED));
```

Comment:
This statement, as opposed to the previous version says an item is only buildable if its item is WOOD_OR_STONE and wood is disabled and stone is disabled.

It's missing a !, or we return as I prefer to the previous case where things are buildable by default and then are not buildable under certain conditions (the condition here being they are not buildable if their are set to WOOD_OR_STONE and both are disabled) :)

Reply:
This is why the tests are failing btw.